### PR TITLE
Redirect users away when they have no active selected contest

### PR DESCRIPTION
**Pull request by mvr320**

### CODE REVIEW

File: webapp/src/Twig/TwigExtension.php [hunk 1]
```diff
@@ -206,6 +206,7 @@ public function getGlobals()
                 }, Countries::getAlpha3Codes())
             ),
             'show_shadow_differences' => $this->tokenStorage->getToken() &&
+                                         !is_null($this->dj->getCurrentContest()) &&
```
Comment (nickygerritsen) [inline webapp/src/Twig/TwigExtension.php:209]:
```suggestion
                                         $this->dj->getCurrentContest() &&
```
sounds shorter

File: webapp/src/Controller/Jury/JuryMiscController.php [hunk 2]
```diff
@@ -62,7 +62,7 @@ public function __construct(EntityManagerInterface $entityManager, DOMJudgeServi
     public function indexAction(Request $request): Response
     {
         $errors = [];
-        return $this->render('jury/index.html.twig', ['errors' => $errors]);
+        return $this->render('jury/index.html.twig', ['errors' => $errors, 'contest_selected' => !is_null($this->dj->getCurrentContest())]);
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/JuryMiscController.php:65]: Are we using this?
Also: while we are at it, maybe hide the analytics and scoreboard links on the jury interface? Should be able to check `current_contest` in twig to see if there is one

File: webapp/src/Controller/Jury/ShadowDifferencesController.php [hunk 3]
```diff
@@ -83,6 +83,11 @@ public function indexAction(Request $request)
             return $this->redirectToRoute('jury_index');
         }
 
+        if (is_null($this->dj->getCurrentContest())) {
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/ShadowDifferencesController.php:86]:
```suggestion
        if (!$this->dj->getCurrentContest()) {
```
here as well?

File: webapp/src/Controller/Jury/JuryMiscController.php (hunk 2 above)
Comment (mvr320) [reply webapp/src/Controller/Jury/JuryMiscController.php:65]: I thought about that and my argumentation to leave it like this was that buttons not being there is very annoying. We could make them gray but UX wise I think people finding out they should press the button [change contest] or not showing the button has the same level of frustration.

So I'm fine with your option, if you indeed prefer it over the redirects.

Comment (nickygerritsen) [reply webapp/src/Controller/Jury/JuryMiscController.php:65]: OK fair point, buttons and links moving around is also a mess. Let's keep them

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**